paasta_tools/cli/schemas/kubernetes_schema.json

```diff
@@ -486,6 +486,20 @@
                         }
                     }
                 },
+                "datastore_credentials": {
```

Comment:
@nemacysts how does this look? I didn't include 'cassandra' since that hasnt been cepped yet. but additionalProperties: True should account for it ?

Reply:
also, all databases begin with an alphabetical character before going alphanumerical + underscores, so let me know if the pattern looks good

Reply:
spoke with @flopex who mentioned it should just be [a-zA-Z0-9_.-]+

Reply:
++ to strictness

>  are we envisioning that those need to do anything substantially different from the existing code? 
nope, will work seamlessly. they'll just have to add their associated key here to the schema. they *may* choose to do things differently though (eg vault paths), so I kept it largely open-ended as a proof of concept for NoSQL/other datastores but ++ to leaving it out then! ty

Reply:
resolving - feel free to bring up again